daphne/src/vdaf/mod.rs

```diff
@@ -932,67 +1028,107 @@ impl VdafConfig {
                 }
             };
 
-            let res = match self {
-                Self::Prio3(prio3_config) => prio3_prep_finish_from_shares(
-                    prio3_config,
-                    0,
-                    leader_step,
-                    leader_message,
-                    helper_message,
-                ),
-                Self::Prio2 { dimension } => prio2_prep_finish_from_shares(
-                    *dimension,
-                    leader_step,
-                    leader_message,
-                    helper_message,
-                ),
-            };
+            match task_config.version {
```

Comment:
Same thing about splitting into two functions here, but since there is quite a bit of common code before this point, you can split into two inner functions (inside this function) that way it doesn't get lost at the bottom of the file

Reply:
Done. I ended up finding a bug above which, after fixing, result in less duplicated code. I decided to split the function by version just as I did above.